File: algorithms/bubble_sort/swift/bubble_sort.swift
func bubbleSort(unsortedArray: Array<NSNumber>) -> Array<NSNumber> {
    //mutable copy of the initial array
    var mutableArray = unsortedArray
    //double looping through the array
    for i in 0..<mutableArray.count {
        for j in 0..<(mutableArray.count - i - 1) {
            //checking the elements
            if (mutableArray[j] as! Int > mutableArray[j + 1] as! Int) {
                let temp = mutableArray[j] //storing into a temp value
                //swapping the elements
                mutableArray[j] = mutableArray[j + 1]
                mutableArray[j + 1] = temp
            }
        }
    }
    //returning the sorted array
    let sortedArray = mutableArray
    return sortedArray
}

File: algorithms/bubble_sort/javascript/bubble-sort.js
function bubbleSort(a)
{
    let swapped;
    do {
        swapped = false;
        for (let i=0; i < a.length-1; i++) {
            if (a[i] > a[i+1]) {
                let temp = a[i];
                a[i] = a[i+1];
                a[i+1] = temp;
                swapped = true;
            }
        }
    } while (swapped);
    return a;
}
